src/Moq/InvocationShape.cs

```diff
@@ -77,6 +77,11 @@ private bool IsOverride(MethodInfo invocationMethod)
 				return false;
 			}
 
+			if (!invocationMethod.IsPublic && method.DeclaringType.GetTypeInfo().IsInterface)
```

Comment:
Hi. This might not be a good solution because I think it doesn't solve the problem in general but just one specific case. For example, this scenario will fail:

```
public interface IExplicit
{
	int Bar();
}

public class Explicit : IExplicit
{
	int IExplicit.Bar() => 1;

	public virtual int Bar() => 2;
}

[Fact]
public void ExplicitTest()
{
	var mock = new Mock<Explicit>();
	mock.As<IExplicit>().Setup(m => m.Bar()).CallBase();

	var a = ((IExplicit)mock.Object).Bar();
	var b = mock.Object.Bar();

	Assert.True(a == 1, "a");
	Assert.True(b == 0, "b");
}
```

Assert b will fail.

Reply:
Yes, I got it.
Do you know how I detect this case, when we have two method with equal name, argument and return type?

Cheers!

Reply:
I proposed a different solution at #657. Take a look, I actually used your tests to verify it :bowtie:

Reply:
> Yes, I got it.
> Do you know how I detect this case, when we have two method with equal name, argument and return type?
> 
> Cheers!

In this case, you have two methods one is explicitly implemented and the other is a class method. So whenever the interface method has been invoked you should look if that method is explicitly implemented and if so match with that one instead of matching with the class method. In the case method has not been explicitly implemented you could match with the class method...

However, finding if the method has been explicitly implemented is, IMO, a bit awkward...

Reply:
Yes, is very difficult detect this case, but I was thinking about this case and I think I can detect the explicit implementation.

Cheers!